### ISSUE TITLE
ML.NET aborted with "Training cancelled"

### ISSUE THREAD

**Issue by rustd** [dotnet/machinelearning-modelbuilder#284]
@CBrauer commented on [Sat Oct 12 2019](https://github.com/dotnet/machinelearning/issues/4327)

I tried to do a regression model using the latest ML.NET and Visual Studio 2019.
Here is a screen shot:

### Issue
![image](https://user-images.githubusercontent.com/1317234/66710251-75f5dd00-ed29-11e9-8716-e715a96650bd.png)

My dataset is located at:
https://github.com/CBrauer/CypressPoint.github.io/blob/master/rocket-train-regress.zip.

I don't know where the log file is.  It's not my my Project directory.  Why not?

Charles

---

@codemzs commented on [Mon Oct 14 2019](https://github.com/dotnet/machinelearning/issues/4327#issuecomment-541796748)

@CBrauer I believe this is because AutoML ran out of the allotted time for finding the best model. @justinormont can confirm the same. 

---

@CBrauer commented on [Mon Oct 14 2019](https://github.com/dotnet/machinelearning/issues/4327#issuecomment-541867573)

That's not true.  The max time was 14400.  Here is my setup screen:
![screen2](https://user-images.githubusercontent.com/1317234/66776904-676c0a80-ee7c-11e9-8f37-7a92f91cf6cf.png)

**Comment by JakeRadMSFT** [comment]
@justinormont thoughts?

@LittleLittleCloud can you look into this?

**Comment by LittleLittleCloud** [comment]
KK

**Comment by justinormont** [comment]
The wording "Training cancelled" is from ModelBuilder; I think its use is to indicate the user cancelled the run. Is there another way to throw a `OperationCanceledException`?

**Comment by LittleLittleCloud** [comment]
> KK
> I am able to reproduce the bug

It might be ServiceHub's feature, after I set `cancellationToken` to `None` in `AutoMLServiceFactory`, the training doesn't cancel any more.

![image](https://user-images.githubusercontent.com/16876986/67132892-33b41c00-f1bf-11e9-9048-f1a7c863c8be.png)


 @AArnott Is that the cause for the cancellation? (that serviceHub will cancel the service after a default timeout)

**Comment by AArnott** [comment]
@RyanToth3 would be the best one to comment about default timeouts in requesting servicehub services.

**Comment by RyanToth3** [comment]
Could you share the logs under %temp%/servicehub/logs? There should be something in there telling us what exactly is timing out.

**Comment by CBrauer** [comment]
I would be glad to, but I don’t know which log.  My log directory looks like:

 



 

Since I’m using Visual Studio 2019, it sure would be nice if you guys would put the log file in my Project directory.

 

Charles

**Comment by CBrauer** [comment]
Hi Ryan,

 

I have attached my log directory.

 

Charles

**Comment by RyanToth3** [comment]
@CBrauer I'm not seeing the logs you attached, could you attach them directly to this issue?

**Comment by CBrauer** [comment]
Here are my logs:

[logs.zip](https://github.com/dotnet/machinelearning-modelbuilder/files/3763762/logs.zip)

**Comment by CBrauer** [comment]
Hello Ryan,

 

I uploaded my log directory.

 

Charles

**Comment by RyanToth3** [comment]
@LittleLittleCloud I'm not seeing any errors in getting the ML ServiceHub service in these logs. Is that service request line the one where you're seeing the cancellation being thrown from? What does it mean that the train was cancelled after 3000 iteration? That makes it sound like the ServiceHub service has started and you were already communicating with it. If that's the case ServiceHub itself would not be cancelling anything and the cancellation must be coming from the ML service.

**Comment by LittleLittleCloud** [comment]
Hey Ryan

 

Sorry for the late reply, Here’s the log

 

Regards

Xiaoyun

**Comment by RyanToth3** [comment]
@LittleLittleCloud I'm not seeing the logs you attached, could you attach them directly to this issue? Could you also comment on the rest of my questions regarding your repro of the issue?

**Comment by ShishirKorde-gep** [comment]
I have same issue however am running the get started example of text classification. Using exactly same dataset as attached on the site. I tried with 10 second run as mentioned in the example, and it seems to have failed immediately without any run/iteration in output window.

PS : I ran it on another machine and it worked without issues. So I tried re-installing the feature on machine where it wasn't working but ran into same issue again.

**Comment by LittleLittleCloud** [comment]
Hi @ShishirKorde-gep , we have a fixed build for this, could you try it out to see if this solve your questions? thanks

You can get the download link from #834

**Comment by Jonathan-CG** [comment]
I'm also having the same issue. But it's not exactly the same thing. Training the model using the interface works fine, but using the code generated from the interface after I have trained the model prompts a operation cancelled exception everytime I try to save the model.

At first I thought it was because the file path was wrong, but I tried different ways and I made sure it was right, and even if the file path is correct, everytime I try to save the model, it prompts a operation cancelled exception.

![image](https://user-images.githubusercontent.com/13685010/84995924-97aa2e00-b11a-11ea-9cd6-fd9d8b70adfd.png)

(I'm using .NET Core 3.1)
